server/src/test/java/org/elasticsearch/discovery/DiscoveryDisruptionIT.java

```diff
@@ -59,7 +59,8 @@
 public class DiscoveryDisruptionIT extends AbstractDisruptionTestCase {
 
     public void testIsolatedUnicastNodes() throws Exception {
-        List<String> nodes = startCluster(4, -1, new int[]{0});
+        List<String> nodes = startCluster(4, -1);
+        internalCluster().setHostsListContainsOnlyFirstNode(true);
```

Comment:
I think this has to happen before you start the cluster, or else the cluster will start with full knowledge.

Reply:
Right fixing